### DISCUSSION TITLE
Readonly Mode

### DISCUSSION THREAD

**Discussion by ssendev** [dani-garcia/vaultwarden#4780] [Ideas]
Is there a way to configure a readonly mode where it's possible to look at existing passwords but not possible to change anyting.

My usecase is basically a read replica which can be run completely independently and is still accessible even when the primary instance is offline. Where data is copied to maybe once a day from the "live" instance.

But to avoid HA hassles which make the whole setup only more brittle a readonly replica under another address -maybe with a banner that that is the "readonly instance" and if awailable the primary one should be used- would be pretty nice.

Ideally activated with an environment variable so it's just taking a snapshot of the docker volumes copying them over and then setting the environment variable in the replica.

**Comment by BlackDex** [comment]
There is nothing built-in like this. You probably need to make something like this your self.
Also, there is no way to fully make the database read-only. Since we do need to store some tokens etc... when someone logs-in.

**Comment by ssendev** [comment]
https://github.com/dani-garcia/vaultwarden/pull/4786 contains an implementation of this but it was deemed to be too niche and invasive.

On the plus side, when password availability is the only concern it's possible to use a dedicated client which allows accessing the passwords even when the server is offline.
